test/files/pos/t2377b/a.scala

```diff
@@ -0,0 +1,3 @@
+object Test {
+  (new Q.Inner).foo // make sure foo's type is well-formed
```

Comment:
Indeed, looks like we're binding to the outer `Builder` before the fallback case triggers

```
qscalac -Xprint:typer -uniqid test/files/pos/t2377b/*
[[syntax trees at end of                     typer]] // Q.java
package <empty> {
  object Q#6192 extends  {
    def <init>#14019();
    object Builder#14020 extends  {
      def <init>()
    };
    <static> class Builder#14022 extends _root_.java.lang.Object {
      def <init>()
    };
    object Inner#14023 extends  {
      def <init>();
      object Builder extends  {
        def <init>()
      };
      <static> class Builder extends _root_.java.lang.Object {
        def <init>()
      }
    };
    <static> class Inner#14025 extends _root_#5.java#10.lang.Object {
      def <init>#14026(): Q#6193.Inner#14025;
      def foo#14027(): Q#6193.Builder#14022
    }
  };
  class Q#6191 extends _root_.java.lang.Object {
    def <init>()
  }
}

 // a.scala
package <empty>#4 {
  object Test#6198 extends scala#22.AnyRef#1976 {
    def <init>#6708(): Test#6199.type = {
      Test#6199.super.<init>#2308();
      ()
    };
    new Q#6192.Inner#14025().foo#14027()
  }
}
```

```
⚡ tail -n 100 test/files/pos/t2377b/*
==> test/files/pos/t2377b/Q.java <==
public class Q {

  public static class Builder {}

  public static class Inner {
    public static class Builder {}
    public Builder foo() { return new Builder(); } // this line gives an error, that Builder is ambiguous
  }

}

==> test/files/pos/t2377b/a.scala <==
object Test {
  (new Q.Inner).foo.innerMethod // make sure foo's type is well-formed
}
/code/scala on pr/5304*
⚡ qscalac test/files/pos/t2377b/*
test/files/pos/t2377b/a.scala:2: error: value innerMethod is not a member of Q.Builder
  (new Q.Inner).foo.innerMethod // make sure foo's type is well-formed
                    ^
one error found
```